raw_linux.go

```diff
@@ -310,38 +281,84 @@ func (p *packetConn) handleStats(s unix.TpacketStats) *Stats {
 // sysSocket is the default socket implementation.  It makes use of
 // Linux-specific system calls to handle raw socket functionality.
 type sysSocket struct {
-	fd int
+	f  *os.File
+	rc syscall.RawConn
+}
+
+func (s *sysSocket) SetDeadline(t time.Time) error {
+	return s.f.SetDeadline(t)
+}
+
+func (s *sysSocket) SetReadDeadline(t time.Time) error {
+	return s.f.SetReadDeadline(t)
+}
+
+func (s *sysSocket) SetWriteDeadline(t time.Time) error {
+	return s.f.SetWriteDeadline(t)
+}
+
+func (s *sysSocket) Bind(sa unix.Sockaddr) error {
+	var err error
+	cerr := s.rc.Control(func(fd uintptr) {
+		err = unix.Bind(int(fd), sa)
+	})
+	if err != nil {
```

Comment:
My other project (https://github.com/mdlayher/netlink/blob/master/conn_linux.go#L458) checks the control error first, but I guess I'm not really sure if it matters or not.

Reply:
my reading of the code is they're mutually exclusive errors, so i'm not sure it matters